pytest_doctestplus/plugin.py

```diff
@@ -205,40 +204,39 @@ def collect(self):
                         if ignore_warnings_context_needed:
                             test.examples.insert(0, doctest.Example(source=IGNORE_WARNINGS_CONTEXT, want=''))
 
-                    yield doctest_plugin.DoctestItem(
-                        test.name, self, runner, test)
+                    try:
+                        yield doctest_plugin.DoctestItem.from_parent(
+                            self, name=test.name, runner=runner, dtest=test
+                        )
+                    except AttributeError:
+                        # pytest < 5.4
+                        yield doctest_plugin.DoctestItem(
+                            test.name, self, runner, test)
 
-    class DocTestTextfilePlus(doctest_plugin.DoctestItem, pytest.Module):
-        # Some pytest plugins such as hypothesis try and access the 'obj'
-        # attribute, and by default this returns an error for this class
-        # so we override it here to avoid any issues.
-        def obj(self):
-            pass
+    class DocTestTextfilePlus(pytest.Module):
 
-        def runtest(self):
-            # satisfy `FixtureRequest` constructor...
-            self.funcargs = {}
-            fixture_request = doctest_plugin._setup_fixtures(self)
-
-            options = get_optionflags(self) | FIX
+        def collect(self):
+            encoding = self.config.getini("doctest_encoding")
+            text = self.fspath.read_text(encoding)
+            filename = str(self.fspath)
+            name = self.fspath.basename
+            globs = {"__name__": "__main__"}
 
-            doctest.testfile(
-                str(self.fspath), module_relative=False,
-                optionflags=options, parser=DocTestParserPlus(),
-                extraglobs=dict(getfixture=fixture_request.getfixturevalue),
-                raise_on_error=True, verbose=False, encoding='utf-8')
```

Comment:
Ok so the tests differences are due to this removal of `doctest.testfile`, which is because I basically copied what pytest's doctest plugin is doing, to remove the multiple inheritance, and use `collect` instead of `runtest`. I don't why the old code was doing it another way so ... :man_shrugging: